src/passes/SignatureRefining.cpp

```diff
@@ -50,6 +51,28 @@ struct SignatureRefining : public Pass {
   // will not appear in this map.
   std::unordered_map<HeapType, Signature> newSignatures;
 
+  struct Info {
+    // The calls and call_refs.
+    std::vector<Call*> calls;
+    std::vector<CallRef*> callRefs;
+
+    // The relevant drops. In the initial scan we collect all drops, and then
+    // when we merge the information we'll note the drops for each call. We
+    // list pointers to the drops so that we can replace them (we won't need
+    // the drop if we remove all returned values).
+    std::vector<Expression**> drops;
+
+    // A possibly improved LUB for the results.
+    LUBFinder resultsLUB;
+
+    // Normally we can optimize, but some cases prevent a particular signature
+    // type from being changed at all, see below.
+    bool canModify = true;
+
+    // Sometimes we can modify the signature as a whole, but not the results.
+    bool canModifyResults = true;
```

Comment:
(This is the only change (addition) to this structure, aside from moving it to the outer scope so we can use it in a helper function.)